testsuite/tests/basic/constprop.ml.c

```diff
@@ -1,5 +1,6 @@
 (* TEST
    flags = "-pp '${c_preprocessor}'"
+   ocaml_filetype_flag = "-impl"
```

Comment:
Having an OCaml file with a `.c` suffix feels really strange, however unlikely it is to be being edited/viewed in future. Could this instead work by leaving it a `.ml` file but copying it into the test directory as a `.c` file?

Either way, there should be a comment somewhere explaining why this file has to have a `.c` extension.